src/codegen/codeparser.cpp

```diff
@@ -81,119 +86,157 @@ void CCodeParser::ParseCFiles(wxString className) {
 }
 
-void CCodeParser::ParseCCode(wxString header, wxString source) {
-	ParseCInclude(header);
-	ParseCClass(header);
+void CppCodeParser::ParseCode(wxString &header, wxString &source) {
+	ParseHeaderFilePreProc(header);
+	ParseHeaderFileClass(header);
 
-	ParseSourceFunctions(source);
+	ParseSourceFileFunctions(source);
 }
 
-void CCodeParser::ParseCInclude(wxString code) {
+void CppCodeParser::ParseHeaderFilePreProc(wxString &code) {
 	int userIncludeEnd;
-	m_userInclude = wxT("");
+	m_userIncludeFilePreProc.Clear();
 
 	// find the begining of the user include
-	int userIncludeStart = code.Find(wxT("//// end generated include"));
-	if (userIncludeStart != wxNOT_FOUND) {
+
+	int userIncludeStart = code.Find(wxT("//// end generated include. Place User Code Below"));
+	if (userIncludeStart != wxNOT_FOUND) 
+	{
+
 		userIncludeStart = code.find(wxT('\n'), userIncludeStart);
-		if (userIncludeStart != wxNOT_FOUND) {
+		if (userIncludeStart != wxNOT_FOUND) 
+		{
 			// find the end of the user include
 			userIncludeEnd = code.find(wxT("\n/** Implementing "), userIncludeStart);
 
-			if (userIncludeEnd != wxNOT_FOUND) {
+			if (userIncludeEnd != wxNOT_FOUND) 
+			{
 				userIncludeStart++;
-				m_userInclude = code.substr(userIncludeStart, userIncludeEnd - userIncludeStart);
+				m_userIncludeFilePreProc = code.substr(userIncludeStart, userIncludeEnd - userIncludeStart);
+			}
+		}
+	}
+}
+
+void CppCodeParser::ParseSourceFilePreProc(wxString &code) {
+	int userIncludeEnd;
+	m_userSourceFilePreProc.Clear();
+
+	// find the begining of the user include
+
+	int userIncludeStart = code.Find(wxT("//// end generated include. Place User Code Below"));
+	if (userIncludeStart != wxNOT_FOUND) 
+	{
+		userIncludeStart = code.find(wxT('\n'), userIncludeStart);
+		if (userIncludeStart != wxNOT_FOUND) 
+		{
+			// find the end of the user include
+			wxString Str = m_className + wxT("::");
+			userIncludeEnd = code.find(Str, userIncludeStart);
+			if (userIncludeEnd != wxNOT_FOUND) 
+			{
+				userIncludeEnd = code.rfind(wxT('\n'), userIncludeEnd);
+				if (userIncludeEnd != wxNOT_FOUND) 
+				{
+					userIncludeStart++;
+					m_userSourceFilePreProc = code.substr(userIncludeStart, userIncludeEnd - userIncludeStart);
+				}
 			}
 		}
 	}
 }
 
-void CCodeParser::ParseCClass(wxString code) {
+void CppCodeParser::ParseHeaderFileClass(wxString &code) {
 	int startClass = code.Find(wxT("class ") + m_className);
 	if (startClass != wxNOT_FOUND) {
 		code = ParseBrackets(code, startClass);
 		if (startClass != wxNOT_FOUND) {
-			ParseCUserMembers(code);
+			ParseHeaderFileUserMembers(code);
 		}
 	}
 }
 
-void CCodeParser::ParseCUserMembers(wxString code) {
-	m_userMemebers = wxT("");
-	int userMembersStart = code.Find(wxT("//// end generated class members"));
+void CppCodeParser::ParseHeaderFileUserMembers(wxString &code) {
+	m_userHeaderFileMembers = wxT("");
+	int userMembersStart = code.Find(wxT("//// end generated class members. Place User Code Below"));
```

Comment:
Only use the wxWidgets String-Methods if there is no std::string variant available. The wxWidets methods start with uppercase letters while the std::string interface ones start with lowercase letters.